### DISCUSSION TITLE
Error with predict()

### DISCUSSION THREAD

**Discussion by dsantiago** [Lightning-AI/pytorch-lightning#7068] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Hi @awaelchli and thanks for your time, as you asked in pull requests, i am pinging you here

For other who see this, it's a discussion about Trainer.predict method where it is running BatchNorm Layers, code is below:

https://colab.research.google.com/drive/1jujP4F_prSmbRz-F_wGfWPTKGOmY5DPE?usp=sharing

What is the problem with my approach?

**Comment by awaelchli** [comment, marked as answer]
Predict takes a dataloader, not a tensor. It still "works" because the trainer just iterates through the batch dimension, but then you get an error later because the input lost the batch dimension, and batch norm doesn't work with batch size 1.

**Reply by awaelchli** [reply]
Change 
`preds = trainer.predict(model, x)`
to 
`preds = trainer.predict(model, test_loader)`

**Reply by dsantiago** [reply]
Thanks, for it, i read the signature of the function, i knew it expectes a dataloader but as it just run with a tensor, i thought it was ok... 
Just to think about, if all functions are typed, i dunno why, passing a tensor, it have worked... will double check code later...

Ohh just to mention, i already closed the issue on this, it's was opened well before all these discussions...

Thanks again for clarifications and effort!
